scanomatic/ui_server/status_api.py

```diff
@@ -1,63 +1,72 @@
-from flask import Flask, jsonify
+from flask import Blueprint, g, jsonify
 
-from ..io.rpc_client import _ClientProxy
+from ..io.rpc_client import get_client
 from .general import convert_path_to_url, json_abort
 
 
-def add_routes(app: Flask, rpc_client: _ClientProxy):
+blueprint = Blueprint('status_api', __name__)
 
-    @app.route("/api/status/<status_type>")
-    @app.route("/api/status/<status_type>/<status_query>")
-    def _status_api(status_type="", status_query=None):
 
-        if status_type == 'server':
-            if rpc_client.online:
-                return jsonify(success=True)
-            else:
-                return json_abort(
-                    503,
-                    reason="System not yet ready, probably starting up",
-                )
-        elif not rpc_client.online:
-            return jsonify(success=False, reason="Server offline")
-        elif status_type == 'queue':
-            return jsonify(queue=rpc_client.get_queue_status())
-        elif status_type == 'scanners':
-            if status_query is None or status_query.lower() == 'all':
-                return jsonify(scanners=rpc_client.get_scanner_status())
-            elif status_query.lower() == 'free':
-                return jsonify(
-                    scanners={
-                        s['socket']: s['scanner_name']
-                        for s in rpc_client.get_scanner_status()
-                        if 'owner' not in s or not s['owner']
-                    },
+@blueprint.before_request
+def get_rpc():
+    if 'rpc' not in g:
+        g.rpc = get_client()  # noqa
```

Comment:
what is the noqa for here, and if it really is needed specify the exact rule that should be disabled never use just noqa

Reply:
if this is a preformance optimization it seems more reasonable to put an lru_cache wrapper around `get_client` rather than putting it on the global `g` as I assume it won't be typed properly when used.

Reply:
It's for muting pylint, removed.